cogs/customreactions.py

```diff
@@ -0,0 +1,1111 @@
+                await message.edit(embed=message.embeds[0])
+                number = 0
+                try:
+                    number_sent_by_user_modifying = False
+                    manual_timeout = time.time() + 60
+                    while not number_sent_by_user_modifying:
+                        msg = await self.bot.wait_for('message',
+                                                      check=number_check,
+                                                      timeout=60)
+                        if msg.author == user_modifying:
+                            number = int(msg.content)
+                            number_sent_by_user_modifying = True
+                            await msg.delete()
+                        else:
+                            if time.time() > manual_timeout:
+                                raise asyncio.TimeoutError
+
+                except asyncio.TimeoutError:
+                    pass
+
+                if number < 1 or number > len(current_list):
+                    if proposals:
+                        title = "Current custom reaction proposals\nClick on 🆗 to approve, reject, edit, or see " \
+                                "more information on one of them (Previous attempt received invalid input or timed out)"
+                    else:
+                        title = "Current custom reactions\nClick on 🆗 to edit or see more information on one of " \
+                                "them (Previous attempt received invalid input or timed out)"
+                    p = Pages(ctx,
+                              msg=message,
+                              item_list=reaction_dict,
+                              title=title,
+                              display_option=(2, 10),
+                              editable_content_emoji="🆗",
+                              return_user_on_edit=True)
+                else:
+                    await information_on_react(message, current_list, number,
+                                               proposals)
+                    if proposals:
+                        title = "Current custom reaction proposals\nClick on 🆗 to approve, reject, edit, or see " \
+                                "more information on one of them"
+                    else:
+                        title = "Current custom reactions\nClick on 🆗 to edit or see more information on one of them"
+
+                    # update dictionary since a custom reaction might have been modified
+                    if proposals:
+                        current_list = self.proposal_list
+                    else:
+                        current_list = self.reaction_list
+
+                    if not current_list:
+                        if proposals:
+                            title = 'There are currently no custom reaction proposals in this server'
+                        else:
+                            title = 'There are currently no custom reactions in this server'
```

Comment:
more duplicate string literals. instead, use (and with a better name than what I just came up with 😛 ):

```python
NO_CUSTOM = "There are currently no custom {} in this server"
NO_CUSTOM.format("reaction proposals")
NO_CUSTOM.format("reactions")
```

Reply:
See https://github.com/idoneam/Canary/pull/273#issuecomment-656869764